core/io/ip.cpp

```diff
@@ -146,31 +155,39 @@ IP::ResolverID IP::resolve_hostname_queue_item(const String &p_hostname, IP::Typ
 			resolver->resolve_queues();
 	}
 
+	resolver->mutex->unlock();
 	return id;
 }
 
 IP::ResolverStatus IP::get_resolve_item_status(ResolverID p_id) const {
 
 	ERR_FAIL_INDEX_V(p_id, IP::RESOLVER_MAX_QUERIES, IP::RESOLVER_STATUS_NONE);
 
-	GLOBAL_LOCK_FUNCTION;
+	resolver->mutex->lock();
 	ERR_FAIL_COND_V(resolver->queue[p_id].status == IP::RESOLVER_STATUS_NONE, IP::RESOLVER_STATUS_NONE);
```

Comment:
Will end up blocking the Mutex forever if `resolver->queue[p_id].status == IP::RESOLVER_STATUS_NONE`